File: app/src/main/java/igrek/songbook/model/lyrics/LyricsChar.java
package igrek.songbook.model.lyrics;

public class LyricsChar {
	
	public String c;
	
	public float width;
	
	public LyricsTextType type;
	
	public LyricsChar(String c, float width, LyricsTextType type) {
		this.c = c;
		this.width = width;
		this.type = type;
	}
	
	@Override
	public String toString() {
		return c;
	}
}

File: app/src/main/java/igrek/songbook/chords/lyrics/wrapper/LyricsChar.kt
package igrek.songbook.chords.lyrics.wrapper

import igrek.songbook.chords.lyrics.model.LyricsTextType
import igrek.songbook.chords.lyrics.model.lineWrapperChar

internal data class LyricsChar(
        val c: Char,
        val type: LyricsTextType,
        var x: Float = 0f,
        var width: Float = 0f,
) {
    override fun toString(): String {
        return when (type) {
            LyricsTextType.REGULAR_TEXT -> "$c"
            LyricsTextType.CHORDS -> "[$c]"
            LyricsTextType.LINEWRAPPER -> lineWrapperChar.toString()
        }
    }
}
